test-src/integration_test.js

```diff
@@ -0,0 +1,151 @@
+const crypto = require('crypto');
+const fs = require('fs');
+const path = require('path');
+const {Artifact} = require('../');
+const {Queue} = require('./mock-queue');
+
+async function writeFileOfSize(size, name) {
+  let bytes = await new Promise((resolve, reject) => {
+    crypto.randomBytes(size, (err, buf) => {
+      if (err) {
+        return reject(err);
+      }
+      return resolve(buf);
+    });
+  });
+  fs.writeFileSync(name, bytes);
+}
+
+function checkFilesEqual(a, b) {
+  a = fs.readFileSync(a);
+  b = fs.readFileSync(b);
+  if (a.length !== b.length) {
+    throw new Error('Files are different sizes');
+  }
+  for (let i = 0 ; i <= a.length ; i++) {
+    if (a[i] !== b[i]) {
+      throw new Error(`Files differ at byte ${i}: ${a[i]} !== ${b[i]}`);
+    }
+  }
```

Comment:
Nope, I'd rather avoid using underscore or lodash when comparing two arrays is so trivial.